### DISCUSSION TITLE
Primitives or Atoms

### DISCUSSION THREAD

**Discussion by AnaMontes11** [Telefonica/mistica-design#912] [Proposals]
Actualmente en la documentación, hablamos de Átomos y en otros de Primitives (storybook por ejemplo). 
De cara a unificar y decidirnos uno, ¿qué término deberíamos usar?

**Comment by yceballost** [comment]
Pueden convivir ambos?

Una construcción atómica me encajaría más que una construcción con primitivas. O igual hay que llamar a átomos a las primitivas el problema es que si llamamos átomos, deberíamos de hablar de moléculas y organismos, por ejemplo a los patterns ya no serían patterns sino organismos

**Comment by aweell** [comment]
Considero que los dos tienen cabida y que no son equivalentes. 

Desde mi punto de vista:

* Todas las primitivas pueden ser átomos
* No todos los átomos son primitivas

Os traigo un bonito recuerdo, clases de filosofía:

Utilizando un diagrama de Euler, A (Primitivas) es un subconjunto de B (Átomos)

![Screenshot 2022-11-18 at 10 57 18](https://user-images.githubusercontent.com/44420072/202674479-518f6543-9355-4f51-9a04-b19cd3deea46.png)



Entendiendo la definición de átomos como:

**Elementos de mística que en combinación con otros (ya sean de la librería o custom) construyen soluciones a necesidades que no cubre el sistema (Construcciones atómicas).**

Y entendiendo la definición de primitivas como:

**Componentes con un alto nivel de abstracción y alta flexibilidad que están destinados reutilizarse de manera habitual en el sistema ya sea para:**

- **Construir componentes con mayor especificidad de manera interna**
- **Proveer de herramientas para agilizar construcciones atómicas.**

**Reply by yceballost** [reply]
me encaja completamente

**Comment by AnaMontes11** [comment]
En muchas ocasiones estos conceptos se entre mezclan y en otras se muestran como entes separados pero dada la sutil diferencia entre ellos me parece interesante ver qué las consideramos en nuestro caso. Comparto el enfoque de entenderlos como un contenido dentro del otro. 
Yo como entiendo las primitivas (tokens) elementos propios de cada marca y por otro lado los átomos (elementos mínimos que no se pueden deconstruir en otros más pequeños comunes a cada marca que componen la librería)
